lib/component_factory.rb

```diff
@@ -78,16 +80,20 @@ def _transform_row(component, inner)
   def _transform_columns(component, inner)
 
     col_count = component.parent.elements.size
-    small_size = component.attributes['small'] || self.column_count
-    large_size = component.attributes['large'] || component.attributes['small'] || (self.column_count / col_count).to_i
+
+    small_val = component.attribute('small') ? component.attribute('small').value : nil
+    large_val = component.attribute('large') ? component.attribute('large').value : nil
+
+    small_size = small_val || self.column_count
+    large_size = large_val || small_val || (self.column_count / col_count).to_i
```

Comment:
@kball alright these changes appear to fix the issue. Also added a spec for when neither small nor large column widths are defined.